src/contracts/GPv2Settlement.sol

```diff
@@ -290,6 +290,23 @@ contract GPv2Settlement {
         filledAmount[uid] = currentFilledAmount;
     }
 
+    /// @dev Allows settlment function to make arbitrary contract executions.
+    /// @param interaction contains address and calldata of the contract interaction.
+    function executeInteraction(GPv2Encoding.Interaction memory interaction)
+        internal
+        returns (bytes memory response)
```

Comment:
Why do we need to return the response bytes? I think everything related to a single contract interaction should happen in this function.

More specifically, how should the settlement contract deal with the presence or absence of response bytes? Any ideas?

Reply:
I am only returning this because of the compiler warning:
```sh
src/contracts/GPv2Settlement.sol:303:9: Warning: Return value of low-level calls not used.
        (interaction.target).delegatecall(interaction.callData);
```

I'm fine to remove this, just wanted to supress the warning in the most natural way.

Reply:
The issue with this is that we will be creating a new bytes array for every call that we probably wont use (which seems a bit wasteful). If there is a way around this, that would be better.

Reply:
I could explicitly not accept the return data at the expense of the warning at compile time.

Reply:
Hmm, Not super important but I wonder if Solidity still copies the return data to memory in that case.

Reply:
You should be able to write `(bool success, ) = (interaction.target).call(interaction.callData);` without getting a warning.